src/sql-parser/src/parser.rs

```diff
@@ -1692,6 +1694,32 @@ impl Parser {
         }))
     }
 
+    fn parse_create_type(&mut self) -> Result<Statement, ParserError> {
+        let name = self.parse_identifier()?;
+        self.expect_keyword("AS")?;
+        self.expect_keyword("MAP")?;
+
+        self.expect_token(&Token::LParen)?;
+
+        self.expect_keyword("KEY_TYPE")?;
+        self.expect_token(&Token::Eq)?;
+        let key_typ = self.parse_data_type_or_custom()?;
+
+        self.expect_token(&Token::Comma)?;
+
+        self.expect_keyword("VALUE_TYPE")?;
+        self.expect_token(&Token::Eq)?;
+        let value_typ = self.parse_data_type_or_custom()?;
```

Comment:
To match Postgres we should accept these in either order. Looking at it more, Postgres supports both:

```
CREATE TYPE foo AS RANGE (subtype = 'text')
CREATE TYPE foo AS RANGE (subtype = text)
```

I.e., where the value is either a literal or ident. I think what's actually going on here is that we want to use standard WITH options here, and then teach WITH options to accept either idents or strings. We already have a lot of code for that via `SetVariableValue`, so I'm thinking we could just use that type in `SqlOption`. Does that make sense?